src/lib/es5.d.ts

```diff
@@ -210,6 +210,8 @@ interface ObjectConstructor {
       * Returns the names of the enumerable properties and methods of an object.
       * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
       */
+    keys<T>(o: Array<T>): string[];
+    keys<T extends { [key: string]: any }>(o: T): (keyof T & string)[];

```

Comment:
These intersections do make the intellisense popups uglier, but they're required for correctness. 

Could the solution to that ugliness be for the compiler to do some work simplifying intersections (and unions)? E.g., `'a' & string` could just be simplified to to `'a'` (or, more generally, `type & subType` could just be simplified to `subType`). Any efforts in that direction could also help with the bug I brought up in #11426. Would this be that hard? I'd kind of expect simplifying logical formulations to be a somewhat solved problem with theorem provers etc, though I haven't looked into it, so maybe that's incredibly naive. Or maybe it's way too slow to apply here in the general case. Still, specific simplification rules might be a good idea.